apps/els_lsp/src/els_completion_provider.erl

```diff
@@ -44,8 +44,17 @@ handle_request({completion, Params}, State) ->
                             }
    , <<"textDocument">> := #{<<"uri">> := Uri}
    } = Params,
-  ok = els_index_buffer:flush(Uri),
-  {ok, #{text := Text} = Document} = els_utils:lookup_document(Uri),
+  {ok, Document} = els_utils:lookup_document(Uri),
+  #{text := Text, buffer := Buffer} = Document,
+  %% Ensure there are no pending changes. This causes a heavy sync
+  %% point, since for a big module indexing could take a while.  Maybe
+  %% just do a 'soft' flush and avoid indexing here?
+  case Buffer of
+    undefined ->
+      ok;
+    _ ->
+      ok = els_buffer_server:flush(Buffer)
```

Comment:
I wonder if we cannot be a bit sloppy here and to make flushing delayed. Looking at the LSP, I see that we can return partial completion results. So what we can do to increase responsiveness is this:
- if no pending changes, just what we do now
- i pending changes, trigger flushing, and return completion results based on current knowledge
- when indexing is complete return full completion results

Reply:
Partial results are not yet implemented, I will add this as a follow up.